gdraw/ggdkcdraw.c

```diff
@@ -285,7 +285,7 @@ static cairo_surface_t *_GGDKDraw_GImage2Surface(GImage *image, GRect *src) {
     /*  premultiply each channel by alpha. We can reuse it for non-transparent*/
     /*  rgb images */
     if (base->image_type == it_true && type == CAIRO_FORMAT_RGB24) {
-        idata = ((uint32 *)(base->data)) + src->y * base->bytes_per_line + src->x;
+        idata = ((uint32 *)(base->data + src->y * base->bytes_per_line)) + src->x;
```

Comment:
This fixes the oob memory access

Reply:
Assuming `base->data` is a byte-sized pointer (e.g. `char*`): This one line of the change looks good to me. I did not look at anything else in here.